### PR TITLE
Document new CSV capabilities

### LINKED ISSUE #1444: Document new CSV capabilities
these new enhancements will most likely need docs updates. https://github.com/influxdata/influxdb/pull/18779

### PR DESCRIPTION
**Pull request by pierwill**
- Document --error-file flag
- Also italicize "n" in --skipHeader flag
- Link from annotated CSV compatibility warning
- Add #concat syntax to extended annotated CSV
- Mention --error-file flag in write-data CSV doc

Closes #1444.

- [x] Tests pass (no build errors)
- [x] Rebased/mergeable

### CODE REVIEW

File: content/influxdb/v2.0/reference/cli/influx/write/_index.md [hunk 1]
```diff
@@ -30,24 +30,24 @@ influx write [command]
...
+| Flag |                    | Description                                                           | Input type | {{< cli/mapped >}}    |
+|:-----|:-------------------|:----------------------------------------------------------------------|:----------:|:----------------------|
+| `-b` | `--bucket`         | Bucket name                                                           | string     | `INFLUX_BUCKET_NAME`  |
+|      | `--bucket-id`      | Bucket ID                                                             | string     | `INFLUX_BUCKET_ID`    |
+|      | `--configs-path`   | Path to `influx` CLI configurations (default `~/.influxdbv2/configs`) | string     | `INFLUX_CONFIGS_PATH` |
+|      | `--debug`          | Output errors to stderr                                               |            |                       |
+|      | `--encoding`       | Character encoding of input (default `UTF-8`)                         | string     |                       |
+|      | `--error-file`     | Path to a file used for recording rejected row errors                 | string     |                       |
```
Comment (pierwill) [inline content/influxdb/v2.0/reference/cli/influx/write/_index.md:40]: This is the relevant change for this PR.

File: content/influxdb/v2.0/reference/cli/influx/write/_index.md [hunk 2]
```diff
@@ -30,24 +30,24 @@ influx write [command]
...
-| `-p` | `--precision`      | Precision of the timestamps (default `ns`)                            | string     | `INFLUX_PRECISION`   |
-|      | `--skipHeader`     | Skip first n rows of input data                                       | integer    |                      |
-|      | `--skipRowOnError` | Output CSV errors to stderr, but continue processing                  |            |                      |
-|      | `--skip-verify`    | Skip TLS certificate verification                                     |            |                      |
-| `-t` | `--token`          | Authentication token                                                  | string     | `INFLUX_TOKEN`       |
-| `-u` | `--url`            | URL to import data from                                               | string     |                      |
+| Flag |                    | Description                                                           | Input type | {{< cli/mapped >}}    |
+|:-----|:-------------------|:----------------------------------------------------------------------|:----------:|:----------------------|
```
Comment (kelseiv) [inline content/influxdb/v2.0/reference/cli/influx/write/_index.md:34]: looks like `--active-config` disappeared

File: content/influxdb/v2.0/reference/syntax/annotated-csv/extended.md [hunk 3]
```diff
@@ -325,6 +326,18 @@ Use the `±HHmm` format to specify the timezone offset relative to UTC.
...
+
+The `#concat` annotation adds a new column that is concatenated from existing columns according to bash-like string interpolation literal with variables referencing existing column labels.
+
+For example:
+
+```
+#concat,string,fullName,${firstName} ${lastName}
+```
+
+This is especially useful when constructing a timestamp from multiple columns.
```
Comment (kelseiv) [inline content/influxdb/v2.0/reference/syntax/annotated-csv/extended.md:339]: Shall we add an example similar to the one that Russ demo'd? Or example here: https://github.com/influxdata/influxdb/issues/18741

File: content/influxdb/v2.0/write-data/developer-tools/csv.md [hunk 4]
```diff
@@ -226,6 +226,8 @@ influx write -b example-bucket \
 Skipped rows are ignored and are not written to InfluxDB.
 {{% /warn %}}
 
+Use the `--error-file` flag to record errors to a file.
```
Comment (kelseiv) [inline content/influxdb/v2.0/write-data/developer-tools/csv.md:229]: Perhaps we can add example that describe how this error file contains all rows that can't be imported and includes comments about what is wrong, for example:

# error : line 3: column 'a': '1.1' cannot fit into long data type
cpu,1.1

**Comment by kelseiv** [comment]
Let's mention the parsing `long` or `unsignedLong` data types issue on this page https://docs.influxdata.com/influxdb/v2.0/reference/syntax/annotated-csv/extended

Details about how the CSV fails when parsing `long` or `unsignedLong` data types in strict mode. Otherwise, the data is truncated to omit the fractional value and displays a message identifying the line and column that was truncated.

Looks like we can add an example of how this works here too: https://docs.influxdata.com/influxdb/v2.0/write-xdata/developer-tools/csv/

File: content/influxdb/v2.0/reference/cli/influx/write/_index.md (hunk 2 above)
Comment (pierwill) [reply content/influxdb/v2.0/reference/cli/influx/write/_index.md:34]: I don't see --active-config listed in the currently live docs: https://docs.influxdata.com/influxdb/v2.0/reference/cli/influx/write/.

Comment (kelseiv) [reply content/influxdb/v2.0/reference/cli/influx/write/_index.md:34]: Gotcha, the --active-config flag is part of RC added to beta-17 branch. We'll want to retain when resolving this merge conflict/ pushing changes to beta-17.

**Review by kelseiv — APPROVED** [review]

**Pull request merged**